skbio/io/format/embedding.py

```diff
@@ -0,0 +1,212 @@
+r"""Embedding format (:mod:`skbio.io.format.embed`).
+====================================================
+
+.. currentmodule:: skbio.io.format.embed
+
+This module provides support for reading and writing embedding files that
+are outputted by sequential language models such as protein language models
+(pLMs).
+
+Format Support
+--------------
+**Has Sniffer: Yes**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |Yes   |generator of :mod:`skbio.embedding.ProteinEmbedding` objects   |
++------+------+---------------------------------------------------------------+
+|Yes   |Yes   |:mod:`skbio.embedding.ProteinEmbedding` objects                |
++------+------+---------------------------------------------------------------+
+
+Format Specification
```

Comment:
Could the format spec include dimension and dtype expectations? 

It may be nice to include a `.attr['format']` and `.attr['format-version']` in the file

Reply:
`.attr` is formally part of HDF5

Reply:
Yes adding dtype information could be extremely useful for dealing with 16/8/4-bit precision tensors.  I'm not as sure about the dimensionality, since that can be explicitly inferred from the embeddings.

Reply:
This may be a challenge point. `bfloat16` for example is I believe still in [RFC](https://forum.hdfgroup.org/t/hdf5-rfc-adding-support-for-16-bit-floating-point-and-complex-number-datatypes-to-hdf5/11975) stage for HDF5